src/wallet3/transaction_scanner.cpp

```diff
@@ -34,6 +34,7 @@ namespace wallet
 
     // A derivation is simply the private view key multiplied by the tx public key
     // do this for every tx public key in the transaction
+    //TODO SEAN THIS IS A BOTTLENECK
```

Comment:
thinking here: what if we used a threadpool kind of entity where we put heavy but pure jobs onto. then have the these heavy functions (if they can be pure) return a `std::future` that will be fulfilled when the operation is done in the thread pool. we could have those fan out onto as many threads as we want.

the following could be something looking like:
```
auto ftr_derivations = wallet_keys->async_generate_key_derivation(tx_public_keys);
auto derivation = ftr_derivations.get();
```